fdbserver/Resolver.actor.cpp

```diff
@@ -169,10 +171,28 @@ struct Resolver : ReferenceCounted<Resolver> {
 	Counter splitRequests;
 	int numLogs;
 
+	// End-to-end server latency of resolver requests.
+	LatencySample resolverLatency;
+
+	// Queue wait times, per request.
+	LatencySample queueWaitLatency;
+
+	// Actual work, per req request.
+	LatencySample computeLatency;
+
+	// Maximum number of waiters in queue during metrics interval.
+	int64_t maxQueueDepth;
```

Comment:
Can this just be a `Counter`? That way you get the reset to 0 behavior for free as part of the `CounterCollection` and don't need the `getAndReset..` function.

Reply:
Are you able to set the value of a Counter? I only saw methods to increment.

Also, when resetting, I need to reset this value to the number of waiters in the queue, not 0.

If this is all reasonable to do with a Counter, then sure.

Reply:
You're right, `Counter` can only be incremented, not set. Presumably the functionality *should* exist in the `Counter` interface, since this field is still counting things, but doesn't as is.